components/calendar/src/chinese.rs

```diff
@@ -176,11 +161,11 @@ impl Calendar for Chinese {
     //Count the number of months in a given year, specified by providing a date
     // from that year
     fn days_in_year(&self, date: &Self::DateInner) -> u16 {
-        Self::days_in_provided_year(date.0 .0.year)
+        date.0.days_in_year_inner()
     }
 
     fn days_in_month(&self, date: &Self::DateInner) -> u8 {
-        Self::month_days(date.0 .0.year, date.0 .0.month)
+        date.0.days_in_month_inner()
     }
 
     #[doc(hidden)] // unstable
```

Comment:
Do we have plans on how to address `offset_date` which takes a `&mut Self::DateInner` and therefore may need to update the cache?

Reply:
Yes

Reply:
well, since we have the chance now I think this code should update the cached value in the offset_date impl. But yes, we have a plan on how to handle caching there